doc/classes/GPUParticles3D.xml

```diff
@@ -150,8 +150,8 @@
 	<signals>
 		<signal name="finished">
 			<description>
-				Emitted when all active particles have finished processing. When [member one_shot] is disabled, particles will process continuously, so this is never emitted.
-				[b]Note:[/b] Due to the particles being computed on the GPU there might be a delay before the signal gets emitted.
+				Emitted when all active particles are no longer visible. To immediately emit new particles, call the emitter with [code]restart()[/code] as the emitter may still have [code]emitting == true[/code] (due to invisible particles still being processed).
```

Comment:
> Emitted when all active particles are no longer visible

I don't think this is accurate. Visibility has nothing to do with this signal. You can make your node invisible and it won't fire the signal.

Reply:
You say "visibility has nothing to do with this signal" - and that "make your node invisible and won't fire"? Sounds like visibility does have something to do with the signal?

Reply:
The problem here is, that `visibility` already has a rather clear meaning in the godot context. (The property on nodes).

"Are no longer drawn" / "Are no longer displayed" would convey the same idea without using the word `visibility`.

Reply:
I'm saying that even if you make you node invisible (and thus make all the particles no longer visible) the signal won't fire. Precisely because the signal is not connected to visibility whatsoever. 

The signal fires after all the nodes have finished processing (regardless of whether the particles are visible or not). But, as the current iteration of the docs points out, it's inexact due to timing issues between the CPU and the GPU.

Again, visibility does not play a factor.

Reply:
I believe this is resolved. Thanks for the help!